reVX/setbacks/base.py

```diff
@@ -0,0 +1,658 @@
+        -------
+        float
+        """
+        if self.multiplier is None:
+            setback = None
+        else:
+            setback = self.plant_height * self.multiplier
+
+        return setback
+
+    @property
+    def multiplier(self):
+        """Generic setback multiplier.
+
+        Returns
+        -------
+        int | float
+        """
+        return self._multi
+
+    @property
+    def arr_shape(self):
+        """Rasterize array shape.
+
+        Returns
+        -------
+        tuple
+        """
+        return self._shape
+
+    @property
+    def profile(self):
+        """Geotiff profile.
+
+        Returns
+        -------
+        dict
+        """
+        return self._profile
+
+    @property
+    def crs(self):
+        """Coordinate reference system.
+
+        Returns
+        -------
+        str
+        """
+        return self.profile['crs']
+
+    @property
+    def regulations(self):
+        """Regulations table.
+
+        Returns
+        -------
+        geopandas.GeoDataFrame | None
+        """
+        return self._regulations
+
+    def _parse_features(self, features_fpath):
+        """Abstract method to parse features.
+
+        Parameters
+        ----------
+        features_fpath : str
+            Path to file containing features to setback from.
+
+        Returns
+        -------
+        geopandas.GeoDataFrame
+            Geometries of features to setback from in exclusion coordinate
+            system
+        """
+        return gpd.read_file(features_fpath).to_crs(crs=self.crs)
+
+    # pylint: disable=unused-argument
+    def _check_regulations(self, features_fpath):
+        """Reduce regulations to state corresponding to features_fpath.
+
+        Parameters
+        ----------
+        features_fpath : str
+            Path to shape file with features to compute setbacks from.
+
+        Returns
+        -------
+        regulations : geopandas.GeoDataFrame | None
+            Regulations table.
+        """
+        logger.debug('Computing setbacks for regulations in {} counties'
+                     .format(len(self.regulations)))
+
+        return self.regulations
+
+    def _get_setback(self, county_regulations):
+        """Compute the setback distance for the county.
+
+        Compute the setback distance (in meters) from the
+        county regulations or the plant height.
+
+        Parameters
+        ----------
+        county_regulations : pandas.Series
+            Pandas Series with regulations for a single county
+            or feature type.
+
+        Returns
+        -------
+        setback : float | None
+            setback distance in meters, None if the setback "Value Type"
+            was not recognized
+        """
+
+        setback_type = county_regulations['Value Type'].strip()
+        setback = float(county_regulations['Value'])
+        if setback_type == 'Maximum Structure Height':
+            setback *= self.plant_height
```

Comment:
if the setback type is maximum structure height, isnt the setback just equivelant to the plant height? I would think we only multiply by the plant height if value type is "Max-tip Height Multiplier" not "Maximum Structure Height"

Reply:
So I explicity checked with Anthony and Yuying, and they said that for the solar csv file, a value type of `Maximum Structure Height` represents a multipler. I can totally see how that would be confusing in the base class however, so I will move that to the solar setbacks subclass and keep this one generic.

Reply:
Oops still WIP

Reply:
Yeah might be a good place for an abstract method. That is super confusing about the multiplier.... we should touch base with anthony about nomenclature.

Reply:
Good point, I will throw that in

Reply:
Okay here's the plan: lets do the following options (plus any that you think are necessary) for solar: "meters" and "Structure Height Multipler". We should also do the following:

1. Make `_get_setback()` -> `get_regulation_setback()`
2. Add in the docstrings what the options are (Max-tip height multiplier, meters, etc...) No need for description, these should be self-documenting 
3. Make this `setback_type.lower()` so its not case-sensitive

Reply:
OK, I think I got everything but definitely let me know if I've overlooked anything